yt_dlp/extractor/tenplay.py

```diff
@@ -1,70 +1,76 @@
+        username, password = self._get_login_info()
+        data = self._download_json('https://10play.com.au/api/user/auth', video_id, 'Getting bearer token', headers={
+            'X-Network-Ten-Auth': _auth_header,
+        }, data=urlencode_postdata({
+            'email': username,
+            'password': password,
+        }))
+        return f"Bearer {data['jwt']['accessToken']}"
 
     def _real_extract(self, url):
         content_id = self._match_id(url)
+        _token = self._get_bearer_token(content_id)
         data = self._download_json(
-            'https://10play.com.au/api/video/' + content_id, content_id)
-        video = data.get('video') or {}
-        metadata = data.get('metaData') or {}
-        brightcove_id = video.get('videoId') or metadata['showContentVideoId']
-        # brightcove_url = smuggle_url(
-        #     self.BRIGHTCOVE_URL_TEMPLATE % brightcove_id,
-        #     {'geo_countries': ['AU']})
+            'https://10play.com.au/api/v1/videos/' + content_id, content_id)
+        _video_url = self._download_json(data.get('playbackApiEndpoint'), content_id, 'Downloading video JSON', headers={'Authorization': _token}).get('source')
         m3u8_url = self._request_webpage(HEADRequest(
-            self._FASTLY_URL_TEMPL % brightcove_id), brightcove_id).geturl()
+            _video_url), content_id).geturl()
         if '10play-not-in-oz' in m3u8_url:
             self.raise_geo_restricted(countries=['AU'])
-        formats = self._extract_m3u8_formats(m3u8_url, brightcove_id, 'mp4')
+        formats = self._extract_m3u8_formats(m3u8_url, content_id, 'mp4')
         self._sort_formats(formats)
 
         return {
-            # '_type': 'url_transparent',
-            # 'url': brightcove_url,
             'formats': formats,
-            'id': brightcove_id,
-            'title': video.get('title') or metadata.get('pageContentName') or metadata['showContentName'],
-            'description': video.get('description'),
-            'age_limit': parse_age_limit(video.get('showRatingClassification') or metadata.get('showProgramClassification')),
-            'series': metadata.get('showName'),
-            'season': metadata.get('showContentSeason'),
-            'timestamp': parse_iso8601(metadata.get('contentPublishDate') or metadata.get('pageContentPublishDate')),
-            'thumbnail': video.get('poster'),
-            'uploader_id': '2199827728001',
-            # 'ie_key': 'BrightcoveNew',
+            'id': content_id,
+            'title': data.get('title'),
+            'description': data.get('description'),
+            'age_limit': parse_age_limit(data.get('classification')),
+            'series': data.get('showName'),
+            'season': data.get('showContentSeason'),
+            'timestamp': data.get('published'),
+            'thumbnail': data.get('imageUrl'),
+            'uploader_id': 'Channel 10',
```

Comment:
please explain

Reply:
I can put `Channel 10` into the `uploader` entry, and dynamically change `uploader_id` to a value extracted from the `m3u8`, as it was incorrectly hard-coded before.